pkg/checktypes/checktypes.go

```diff
@@ -0,0 +1,193 @@
+	}
+	l.Debugf("Loaded checktypes info from url=%s, number of checktypes loaded=%d", u.String(), len(jchecktypes.Checktypes))
+	return checktypes, nil
+}
+
+// checktypesFromCode returns the checktypes info defined as code in a
+// directory. We support two forms of repositories:
+// One that contains only one check. It's indicated by a path that points
+// directly to directory containing the check, for instance "./vulcan-dkim".
+// The other that can point to more that one check. It's indicated by a path
+// pointing to the a directory that has one or more subdirectories containing
+// the code of the checks, for instance path: "./cmd/checks".
+func checktypesFromCode(u *neturl.URL, l log.Logger) ([]Checktype, error) {
+	path := strings.TrimPrefix(u.String(), "code://")
+	l.Debugf("Loading checktypes from code in: %s", path)
+	dirInfo, err := os.Stat(path)
+	if err != nil {
+		return nil, err
+	}
+	if !dirInfo.IsDir() {
+		return nil, fmt.Errorf("a code repository must point to a dir: %s", path)
+	}
+	var ct Checktype
+	// Check first is the dir points directly to a checktype.
+	if ct, err = readChecktype(path, l); err == nil {
+		return []Checktype{ct}, nil
+	}
+
+	if err != errNoChecktypeDir {
+		return nil, err
+	}
+	l.Debugf("Looking for checktypes defined in the subdirs of the dir %s", path)
+	// err==errNoChecktypeDir, so the directory itself does not contain a
+	// checktype. Check if any of its subdirs contains one or more checktypes
+	// definitions.
+	entries, err := os.ReadDir(path)
+	if err != nil {
+		return nil, fmt.Errorf("error reading code checktypes repository %s: %v", path, err)
+	}
+	var checktypes []Checktype
+	for _, e := range entries {
+		if !e.IsDir() {
+			continue
+		}
+		dirpath := filepath.Join(path, e.Name())
+		ct, err := readChecktype(dirpath, l)
+		if err == errNoChecktypeDir {
+			continue
+		}
+		if err != nil {
+			return nil, err
+		}
+		checktypes = append(checktypes, ct)
+	}
+	return checktypes, nil
+}
+
+func readChecktype(dirpath string, l log.Logger) (Checktype, error) {
+	l.Debugf("Looking if the directory %s contains the code of a checktype", dirpath)
+	manifestPath := filepath.Join(dirpath, "manifest.toml")
+	l.Debugf("Trying to read a manifest from %s", manifestPath)
+	m, err := ReadManifest(manifestPath)
+	// We consider a directory to be contain the code of a check if it has
+	// a manifest.
+	if errors.Is(err, fs.ErrNotExist) {
+		l.Debugf("The directory %s doesn't contain the code of a checktype", dirpath)
+		return Checktype{}, errNoChecktypeDir
+	}
+	if err != nil {
+		return Checktype{}, fmt.Errorf("error reading manifest file %s, %v", manifestPath, err)
+	}
+	options, err := m.UnmarshalOptions()
+	if err != nil {
+		return Checktype{}, fmt.Errorf("invalid options in manifest %s: %w", manifestPath, err)
+	}
+	assets, err := m.AssetTypes.Strings()
+	if err != nil {
+		return Checktype{}, fmt.Errorf("invalid asset types in manifest %s: %w", manifestPath, err)
+	}
+	name := path.Base(dirpath)
```

Comment:
In this case I would add a suffix to the name (i.e. `_code`)in order to not collide in case of using two checktypes repositories.

Imagine you want to compare two versions of the same check.

```sh
vulcan-local -t www.google.com -checktypes code://./cmd -checktypes https://adevinta.github.io/vulcan-checks/checktypes/edge.json -i vulcan-certinfo
```

as the name of the check type will be the same only the last will be executed.

Also, more than one code could be specified so this could be solved adding `_code1`, `_code2`, ... 

```sh
vulcan-local -t www.google.com -checktypes code://./cmd -checktypes code://../other/cmd -i vulcan-certinfo
```

Also there is a comment related to the image name that could lead to name collision in those cases.

Reply:
you mean add the suffix ``_code`` to name of the check?, if that's the case, I though that was the desired behaviour, I mean I though this could also happen now when you aren't using code to the define checks. In any case I don't mind changing it. Regarding the image collision names, do you think we can wait to fix them in a future PR? just to limit a little bit the scope of this one.

Reply:
Yep, we can postpone it.